### ISSUE TITLE
Failure running under windows when there isn't an entry in .git/refs/heads for this branch

### ISSUE THREAD

**Issue by nikospara** [kurttheviking/git-rev-sync-js#11]
When running under Windows and there isn't an entry in `.git/refs/heads` for this branch, invoking `git-rev-sync` produces the following error:

```
    ref = packfileRegex.exec(packfileContents)[1];
                                              ^
TypeError: Cannot read property '1' of null
    at long (C:\TEST\node_modules\git-rev-sync\index.js:95:47)
```
### Steps to reproduce
1. Create a new directory (e.g. `C:\TEST` as above), run `git init`.
2. Add an `.npignore` with the following contents:
   
   ```
   /node_modules/
   ```
3. Add an `index.js` with the following contents:
   
   ``` js
   var grs = require('git-rev-sync');
   console.log(grs.short());
   ```
4. Add the previous files and commit the initial commit.
5. Run `npm install git-rev-sync`.
6. Now make a few branches: `git branch aaa`, `git branch bbb`.
   - Running `node index.js` at this point should work.
7. Run `git pack-refs --all` (this deletes the refs from `.git/refs/heads` and places them in the `.git/packed-refs`).
8. Run `node index.js` to observe the failure.
### Proposed solution

Tracking the error, it seems to be caused by [this line](https://github.com/kurttheviking/git-rev-sync/blob/master/index.js#L92):

``` js
var refToFind = path.join('refs', 'heads', b);
```

In Windows, `path.join()` will produce the string `refs\\heads\\BRANCH` - i.e. using backslashes!

Proposed solution: simply use `Array.prototype.join` as:

``` js
var refToFind = ['refs', 'heads', b].join('/');
```

**Comment by kurttheviking** [comment]
Thanks for the bug report! I can take a look later this week and the proposed solution. If you'd like, go ahead and submit a PR and I can merge that in.

**Comment by wojciechpolak** [comment]
I have also encountered the same problem. Please make a new release and npm push with this fix.

**Comment by kurttheviking** [comment]
@nikospara @wojciechpolak -- thanks for the help! I just published v1.7.1 (https://www.npmjs.com/package/git-rev-sync); let me know if you have any problems
